### PR TITLE
Replace ledger-constraints with tx-constraints in use-cases.

### PR DESCRIPTION
**Pull request by ghost**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
    - [ ] Important changes are reflected in changelog.d of the affected packages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [x] Reviewer requested

### CODE REVIEW

**Comment by koslambrou** [comment]
You haven't changed the constraints in the plutus-use-cases. The `mkTxConstraints` that you are using is still using the `mkTxParams` of the old constraint library.  I did the same error when I tried it out previously :D 

You need to duplicate the functions in `Plutus.Contract.Request` so that they work for both the old and the new constraints library. Eventually, we will deprecate the old ones. I suggest the keep the current names for the old version, and create new ones for the new versions. Something simple like `mkTxConstraints'` for the new version which we'll rename back to `mkTxConstraints` once the old version is deleted.

**Review by sjoerdvisscher — APPROVED** [review]

File: plutus-use-cases/test/Spec/Uniswap.hs [hunk 1]
```diff
@@ -625,15 +625,16 @@ prop_Whitelist = checkErrorWhitelist defaultWhitelist
 
 tests :: TestTree
 tests = testGroup "uniswap" [
-    checkPredicateOptions (iterate increaseTransactionLimits defaultCheckOptions !! 4) "can create a liquidity pool and add liquidity"
-        (assertNotDone Uniswap.setupTokens
-                       (Trace.walletInstanceTag w1)
-                       "setupTokens contract should be still running"
-        .&&. assertNoFailedTransactions)
-        Uniswap.uniswapTrace
+    -- TODO: waiting for JSON cardano instances
```
Comment (berewt) [inline plutus-use-cases/test/Spec/Uniswap.hs:628]: Which instances are missing?

**Review by berewt — APPROVED** [review]
👍

File: plutus-use-cases/plutus-use-cases.cabal [hunk 2]
```diff
@@ -98,6 +98,7 @@ library
     , plutus-ledger              >=1.1.0
     , plutus-ledger-constraints  >=1.1.0
```
[the remark this answers is not in the document]
Comment (berewt) [reply plutus-use-cases/plutus-use-cases.cabal:99]: Ok, as we have a ticket to migrate the content of `plutus-ledger-constraints` in `plutus-tx-constraints`, I guess we can do it later then.

**Pull request merged**